### ISSUE TITLE
Is YARP production-ready?

### ISSUE THREAD

**Issue by artdedeco** [microsoft/reverse-proxy#1205]
My team has been eagerly awaiting for a supported/production-ready(ish?) version of YARP for perhaps a year now 😅.

I searched around but couldn't find any information on this. Is YARP officially supported? Do you consider it production-ready? 

If the answer to both is "no", please share your personal experiences if you are using it for production scenarios!

Thanks!

**Comment by jmezach** [comment]
We've been running YARP in production for quite some time now as well and it has been pretty rock-solid so far. Upgrading between previews has been pretty painless as well, except for that one time where we also had to change our configuration due to some breaking changes there. Currently running on the latest preview (including the recently released security patch). We aren't particularly high volume though, although we have peak times of around 2000 RPM.

So to summarise we haven't regretted making the choice to run YARP in production at all.

**Comment by rducom** [comment]
This issue is really interesting.

Can you also share hardware consumption details ? (number of cores, average CPU, and RAM consumption ?)

**Comment by amweiss** [comment]
We run it on k8s on AWS, and a few of the downstream services are rough on keeping connections open driving up CPU a bit. Probably more tuning to be done, bit more but right now we have:

```
limits:
  cpu: 500m
  memory: 256Mi
```

Memory usage is pretty flat at 45% to 60% usage. CPU is volatile from 10% at lowest traffic and 85-95% at peak (until scaling finishes starting a new one).

![image](https://user-images.githubusercontent.com/30146/132027810-eb7827b4-5954-44ba-9036-75bd04a5240e.png)
![image](https://user-images.githubusercontent.com/30146/132028252-69933063-542e-492b-9e85-5b031451e9d2.png)

Then with autoscaling setup, we drop to our min of 3 pods at lowest traffic (in theory this could drop to 1, we just don't allow that), and 6 or 7 at peak.
